lib/logstash_sidekiq_logger.rb

```diff
@@ -0,0 +1,37 @@
+# Sets up a new sidekiq file logger for logstash consumption.
+#
+# To do this it combines data from ActiveJob notifications and from Sidekiq
+# logs.
+#
+# ActiveJob notifications provides data on what job ran with what parameters,
+# this is obtained by setting up a subscription to the ActiveJob instrumentaton
+# that Rails provides.
+#
+# Sidekiq provides no instrumentation so the ActiveJob data is combined on a
+# custom log formatter.
+#
+# The final output looks like this:
+#
+# rubocop:disable LineLength
+# {"job_name":"prison_mailer_request_received","arguments":["gid://prison-visits/Visit/0125d92d-3e27-4835-bc35-7fd5fb75033e"],"queue_name":"mailers","status":"completed","active_job_duration":640.827,"total_duration":691.0,"@timestamp":"2016-06-17T13:22:58.096Z","@version":"1"}
+# {"job_name":"prison_mailer_request_received","arguments":["gid://prison-visits/Visit/0125d92d-3e27-4835-bc35-7fd5fb75033e"],"queue_name":"mailers","status":"to_be_retried","active_job_duration":1.194,"total_duration":4.0,"retry_number":1,"@timestamp":"2016-06-17T13:00:26.298Z","@version":"1"}
+# rubocop:enable LineLength
```

Comment:
I am happy with your suggestion of using the method name.

My idea when writing this is that it could be extracted into a gem eventually and this description would become the Readme.